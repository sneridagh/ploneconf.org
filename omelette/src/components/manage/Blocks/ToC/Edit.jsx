/**
 * Edit toc block.
 * @module components/manage/Blocks/ToC/Edit
 */

import React from 'react';
import PropTypes from 'prop-types';

import View from '@plone/volto/components/manage/Blocks/ToC/View';

/**
 * Edit toc block class.
 * @class Edit
 * @extends Component
 */
const Edit = ({ properties, data }) => (
  <View properties={properties} data={data} />
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
Edit.propTypes = {
  properties: PropTypes.objectOf(PropTypes.any).isRequired,
  data: PropTypes.objectOf(PropTypes.any),
};

export default Edit;
