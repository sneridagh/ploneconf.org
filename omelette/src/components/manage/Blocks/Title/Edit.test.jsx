import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import Edit from './Edit';

const mockStore = configureStore();

global.__SERVER__ = true; // eslint-disable-line no-underscore-dangle

test('renders an edit title block component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <Edit
        properties={{ title: 'My Title' }}
        selected={false}
        block="1234"
        onAddBlock={() => {}}
        onChangeField={() => {}}
        onSelectBlock={() => {}}
        onDeleteBlock={() => {}}
        onFocusPreviousBlock={() => {}}
        onFocusNextBlock={() => {}}
        handleKeyDown={() => {}}
        index={1}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
