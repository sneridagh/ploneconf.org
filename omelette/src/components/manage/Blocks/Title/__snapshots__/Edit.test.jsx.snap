// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an edit title block component 1`] = `<div />`;
